### PR TITLE
Improved Docker image and Docker-Compose deployment

### PR DESCRIPTION
**Pull request by JonathanWilbur**
Please submit your pull requests to the 'dev' branch.

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant (Dojo's code isn't currently flake8 compliant, but we're trying to correct that)
- [N/A] If this is a new feature and not a bug fix, you've included the proper documentation under the /docs folder 

There were a lot of problems with Defect-Dojo that this PR fixes:

- The `Dockerfile` installs MySQL in the same container that Defect-Dojo runs in. This is not a `bug`, per se, but it is bad practice that is not conducive to a microservices architecture.
- To change the settings meant rebuilding the Docker image each time, which was extremely time consuming and tedious.
- Customizing the settings was done by calling `sed` on a settings template (`settings.dist.py`) that contains replaceable strings. This _worked_, but it is bad practice both because of the issue mentioned above as well as the potential for other strings to inadvertently get replaced.

This PR removes `settings.dist.py` and instead uses `settings.py` with the values obtained from environment variables, falling back on defaults if they are not present. This makes building the image faster, since a lot less customization and installation has to be done upfront, and settings can be changed without a rebuild. Further, configuration through environment variables is a common practice in containerized applications, so this new deployment scheme is more consonant with existing practices.

### CODE REVIEW

**Comment by wurstbrot** [comment]
Please be aware that I forked Johanthans repo and added env. variable support for admin password change on first startup.
The pull request should be merged from Johanthan first.

https://github.com/JonathanWilbur/django-DefectDojo/pull/1

**Comment by JonathanWilbur** [comment]
@wurstbrot I will fix the port redirect issue, although that was intentional. My rationale for doing that was to intentionally prevent MITM downgrade attacks made possible by using HTTP, so I just left it commented out in case you want to turn it on. It's not a big deal, though, and it will probably just lead to more confusion than its worth, so I'll fix that.

Also, I have to be honest: I am more of a Docker guy than I am a Django guy. I know very little about Django. Can you give me an idea of what I would have to do to fix your second bullet point? (Also, I would venture to guess that permissions issue is probably an issue with the permissions on the files on your host: they have to map to permissions that are usable by the user in the container.)

**Comment by wurstbrot** [comment]
The user _dojo_ is not allowed to write in the folder _static_, as _static_ is created by an other user. This is a common docker/docker-compose problem.
In case there is not technical solution, the user should be informed in the readme that the folder needs to be created beforehand and the folder rights needs to be adjusted (and how).

Before you do that (in addition to resolving the merge conflicts), I recommend to get the confirmation that this PR will be merged, as the defectdojo-project leaders didn't gave a statement so far and it is a big change.

**Comment by wurstbrot** [comment]
@aaronweaver In case @mtesauro still needs time, is there a way to perform the review process without him?

**Comment by aaronweaver** [comment]
@wurstbrot Prior to this PR I had already added environmental variable support in settings.py in the dev branch. That's what we will use going forward so this PR will need to be adjusted to use that instead.

**Comment by wurstbrot** [comment]
@JonathanWilbur can you perform a merge?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    container_name: 'mysql'
+    environment:
+      # MYSQL_RANDOM_ROOT_PASSWORD: 'yes'
+      MYSQL_ROOT_PASSWORD: 'BLORMPF'
+      MYSQL_DATABASE: 'dojodb'
+      MYSQL_USER: 'dojo'
+      MYSQL_PASSWORD: 'dojo'
+    hostname: 'mysql'
+    # image: 'mysql:latest'
+    image: 'mysql:5.7'
```
[the remark this answers is not in the document]
Comment (BlauerPulli) [reply docker-compose.yml]: i already tried mysql version 8 and get this error: mbind: 'Operation not permitted' see, [here](https://github.com/docker-library/mysql/issues/303). Should be resolved in version 8.0.13.. Till now I tried it just with latest(what is 8.0.14) and got the same error. Maybe pinning the image to 8.0.13 helps in that case..

**Comment by aaronweaver** [comment]
Just an update, testing locally, I've updated the documentation on environment variables [usage](https://defectdojo.readthedocs.io/en/latest/getting-started.html#environment-variables) and for the compose script I'm going to pull out the static password and have that passed in as an environment variable. Will update the docs with that as well once complete.

**Pull request merged**